internal/wazeroir/compiler.go

```diff
@@ -206,10 +206,6 @@ type CompilationResult struct {
 	HasMemory bool
 	// HasTable is true if the module from which this function is compiled has table declaration.
 	HasTable bool
-	// NeedsAccessToDataInstances is true if the function needs access to data instances via memory.init or data.drop instructions.
-	NeedsAccessToDataInstances bool
```

Comment:
this was the culprit -- this is determined by per-function scope here, but was used
in the `compileModuleContextInitialization` which is module scope.